src/utp_stream.cpp

```diff
@@ -3042,11 +3042,11 @@ bool utp_socket_impl::incoming_packet(span<std::uint8_t const> buf
 				// sure to clamp it as a sanity check
 				if (delay > min_rtt) delay = min_rtt;
 
-				do_ledbat(acked_bytes, delay, prev_bytes_in_flight);
-				m_send_delay = delay;
+				do_ledbat(acked_bytes, aux::numeric_cast<int>(delay), prev_bytes_in_flight);
```

Comment:
I'm a little bit worried that these numeric_casts will cause a lot of spurious asserts, as I don't think there are any runtime checks for these values to stay within bounds.